tests/app/service/test_rest.py

```diff
@@ -1348,27 +1332,30 @@ def test_get_detailed_services_groups_by_service(notify_db_session, sample_api_k
     service_1_template = sample_template(service=service_1)
 
     notifications = [
-        create_notification(service_0_template, api_key=api_key_0, status='created'),
-        create_notification(service_1_template, api_key=api_key_1, status='created'),
-        create_notification(service_0_template, api_key=api_key_0, status='delivered'),
-        create_notification(service_0_template, api_key=api_key_0, status='created'),
+        sample_notification(template=service_0_template, api_key=api_key_0, status='created'),
+        sample_notification(template=service_1_template, api_key=api_key_1, status='created'),
+        sample_notification(template=service_0_template, api_key=api_key_0, status='delivered'),
+        sample_notification(template=service_0_template, api_key=api_key_0, status='created'),
     ]
 
     data = get_detailed_services(start_date=datetime.utcnow().date(), end_date=datetime.utcnow().date())
-    data = sorted(data, key=lambda x: x['name'])
+
+    data_dict = {item['id']: item for item in data}
```

Comment:
Please move this to after the following assert.  There's no point creating a dictionary if "data" is the wrong length.

Reply:
Updated.